### PR TITLE
Implement uint128 field encoding for genesis.

### PR DESCRIPTION
**Pull request by arijitAD**
## Changes

- Implement uint128 field encoding for genesis.

## Tests

```

```

## Issues

-

## Primary Reviewer 

- @timwu20

### CODE REVIEW

File: lib/genesis/pallet.go [hunk 1]
```diff
@@ -120,20 +122,19 @@ type contracts struct {
 }
 
 type society struct {
-	Pot        uint64   `json:"Pot"`        //TODO: figure out the type
-	MaxMembers uint32   `json:"MaxMembers"` //TODO: figure out the type
-	Members    []string `json:"Members"`
+	Pot        *scale.Uint128 `json:"Pot"`
+	MaxMembers *scale.Uint128 `json:"MaxMembers"` // TODO: figure out the type
```
Comment (edwardmack) [inline lib/genesis/pallet.go:126]: remove todo

**Review by edwardmack — APPROVED** [review]

**Comment by arijitAD** [comment]
Screenshot 2021-08-17 at 5 45 24 PM
Code coverage is passing for me locally.

File: pkg/scale/uint128.go [hunk 2]
```diff
@@ -153,3 +153,22 @@ func (u *Uint128) trimBytes(b []byte, order binary.ByteOrder) []byte {
 	}
 	return b
 }
+
+// UnmarshalJSON converts data to Uint128.
+func (u *Uint128) UnmarshalJSON(data []byte) error {
+	buf := make([]byte, len(data))
+	copy(buf, data)
```
Comment (EclesioMeloJunior) [inline pkg/scale/uint128.go:160]: the `buf` variable is not being used

**Review by EclesioMeloJunior — CHANGES_REQUESTED** [review]

File: lib/genesis/pallet.go (hunk 1 above)
Comment (arijitAD) [reply lib/genesis/pallet.go]: Done

File: pkg/scale/uint128.go (hunk 2 above)
Comment (arijitAD) [reply pkg/scale/uint128.go]: Done.

**Review by EclesioMeloJunior — APPROVED** [review]
lgtm

**Pull request merged**